artemis-protocols/artemis-mqtt-protocol/src/main/java/org/apache/activemq/artemis/core/protocol/mqtt/MQTTConnectionManager.java

```diff
@@ -112,17 +112,18 @@ synchronized void connect(MqttConnectMessage connect, String validatedUser, Stri
          }
       }
 
+      int defaultMqttSessionExpiryInterval = session.getProtocolManager().getDefaultMqttSessionExpiryInterval();
       MqttProperties connackProperties;
       if (session.getVersion() == MQTTVersion.MQTT_5) {
          session.getConnection().setReceiveMaximum(MQTTUtil.getProperty(Integer.class, connect.variableHeader().properties(), RECEIVE_MAXIMUM, -1));
 
-         sessionState.setClientSessionExpiryInterval(MQTTUtil.getProperty(Integer.class, connect.variableHeader().properties(), SESSION_EXPIRY_INTERVAL, 0));
+         sessionState.setClientSessionExpiryInterval(MQTTUtil.getProperty(Integer.class, connect.variableHeader().properties(), SESSION_EXPIRY_INTERVAL, defaultMqttSessionExpiryInterval));
```

Comment:
@jbertram I have change it according to the documentation

>The default defaultMqttSessionExpiryInterval is -1 which means no clean up will happen for MQTT 3.x clients or for MQTT 5 clients which do not pass their own session expiry interval.


but MQTT 3.1.2.11.2 states

>If the Session Expiry Interval is absent the value 0 is used. If it is set to 0, or is absent, the Session ends when the Network Connection is closed.